File: Shortest un-ordered subarray/Shortest_un-ordered_subarray.py
class Solution:
    def shortestUnorderedSubarray(self, arr):
        increase = False
        decrease = False
        
        for i in range(1, len(arr)):
            if(arr[i-1] < arr[i]):
                increase = True
            if(arr[i-1] > arr[i]):
                decrease = True    
        
        if increase and decrease:
            return 3
        
        return 0 

File: Shortest un-ordered subarray/Shortest_un-ordered_subarray.java
class Solution {

    public int shortestUnorderedSubarray(int arr[]) {
        boolean increase = false;
        boolean decrease = false;
        
        for(int i = 1; i < arr.length; i++){
            if(arr[i-1] < arr[i])
               increase = true;
            if(arr[i-1] > arr[i])
               decrease =true;   
        }
        
        if(increase && decrease)
           return 3;
           
        return 0;   
    }
}
